selfdrive/car/subaru/values.py

```diff
@@ -29,13 +29,33 @@ def __init__(self, CP):
     else:
       self.STEER_MAX = 2047
 
+  RPM_MIN = 0                   # min cruise_rpm
+  RPM_MAX = 3200                # max cruise_rpm
+  RPM_BASE = 600                # cruise_rpm idle, from stock drive
+  RPM_SCALE = 3000              # cruise_rpm, from testing
+
+  THROTTLE_MIN = 0              # min cruise_throttle
+  THROTTLE_MAX = 3400           # max cruise_throttle
+  THROTTLE_BASE = 1810          # cruise_throttle, from stock drive
+  THROTTLE_SCALE = 3000         # from testing
+
+  RPM_DELTA_UP = 50
+  RPM_DELTA_DOWN = 50
+
+  THROTTLE_DELTA_UP = 50
+  THROTTLE_DELTA_DOWN = 50
+
+  BRAKE_MIN = 0
+  BRAKE_MAX = 400
+  BRAKE_SCALE = 1000            # from testing
```

Comment:
can you add these limits to the panda safety?

Reply:
I added the max_limit and longitudinal_allowed checks for relevant brake, rpm and throttle signals